packages/sdk-btc/source/transaction.service.ts

```
@@ -1,55 +1,221 @@
+				},
+				transaction.toHex(),
+			);
 		} catch (error) {
+			console.log(error);
 			throw new Exceptions.CryptoException(error);
 		}
 	}
 
-	@IoC.postConstruct()
-	private onPostConstruct(): void {
-		this.#unspent = new UnspentAggregator({
-			http: this.httpClient,
-			peer: Helpers.randomHostFromConfig(this.configRepository),
+	async #addUtxos(psbt, levels: Levels, accountKey: BIP32Interface, targets, changeAddress): Promise<void> {
+		const feeRate = 55; // satoshis per byte // @TODO Need to get this from endpoint
+		const method = this.#addressingSchema(levels);
+		const id: Services.WalletIdentifier = {
+			type: "extendedPublicKey",
+			value: accountKey.neutered().toBase58(),
+			method: method,
+		};
+
+		const allUnspentTransactionOutputs = await this.unspent.aggregate(id);
+		console.log("allUnspentTransactionOutputs", allUnspentTransactionOutputs);
+
+		const derivationMethod = this.#derivationMethod(method);
+
+		let utxos = allUnspentTransactionOutputs.map((utxo) => {
+			let signingKeysGenerator = addressesAndSigningKeysGenerator(derivationMethod, accountKey);
+			let signingKey: SigningKeys | undefined = undefined;
+			do {
+				const addressAndSigningKey: SigningKeys = signingKeysGenerator.next().value;
+				if (addressAndSigningKey.address === utxo.address) {
+					signingKey = addressAndSigningKey;
+				}
+			} while (signingKey === undefined);
+
+			let extra;
+			if (levels.purpose === 44) {
+				extra = {
+					nonWitnessUtxo: Buffer.from(utxo.raw, "hex"),
+				};
+			} else {
+				const payment = bitcoin.payments.p2sh({
+					redeem: bitcoin.payments.p2wpkh({
+						pubkey: Buffer.from(signingKey.publicKey, "hex"),
+						network: getNetworkConfig(this.configRepository),
+					}),
+					network: getNetworkConfig(this.configRepository),
+				});
+				extra = {
+					witnessUtxo: {
+						script: Buffer.from(utxo.script, "hex"),
+						value: utxo.satoshis,
+					},
+					redeemScript: payment.redeem!.output,
```

Should throw an exception here if `payment.redeem` would ever be nil for any reason.